File: languages/java/src/main/java/com/bloock/sdk/entity/SignatureHeader.java
package com.bloock.sdk.entity;

import com.bloock.sdk.bridge.proto.AuthenticityEntities;

public class SignatureHeader {
  public String getAlg() {
    return alg;
  }

  public String getKid() {
    return kid;
  }

  String alg;
  String kid;

  SignatureHeader(String alg, String kid) {
    this.alg = alg;
    this.kid = kid;
  }

  public static SignatureHeader fromProto(AuthenticityEntities.SignatureHeader header) {
    return new SignatureHeader(header.getAlg(), header.getKid());
  }

  public AuthenticityEntities.SignatureHeader toProto() {
    return AuthenticityEntities.SignatureHeader.newBuilder()
        .setAlg(this.alg)
        .setKid(this.kid)
        .build();
  }
}

File: languages/php/Bloock/Entity/SignatureHeader.php
<?php

namespace Bloock\Entity;

class SignatureHeader
{
    private string $alg;
    private string $kid;

    public function __construct(string $alg, string $kid) {
        $this->alg = $alg;
        $this->kid = $kid;
    }

    public static function fromProto(\Bloock\SignatureHeader $header): SignatureHeader {
        return new SignatureHeader($header->getAlg(), $header->getKid());
    }

    public function toProto(): \Bloock\SignatureHeader {
        $p = new \Bloock\SignatureHeader();
        $p->setAlg($this->alg);
        $p->setKid($this->kid);
        return $p;
    }

    /**
     * @return string
     */
    public function getAlg(): string
    {
        return $this->alg;
    }

    /**
     * @return string
     */
    public function getKid(): string
    {
        return $this->kid;
    }
}